app/src/main/java/com/bagi/flowkotlin/activity/covid19/Covid19Activity.kt

```diff
@@ -2,44 +2,40 @@ package com.bagi.flowkotlin.activity.covid19
 
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
-import android.view.View
 import androidx.databinding.DataBindingUtil
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.Observer
 import androidx.lifecycle.ViewModelProviders
 import androidx.recyclerview.widget.StaggeredGridLayoutManager
 import com.bagi.flowkotlin.R
 import com.bagi.flowkotlin.databinding.ActivityMainBinding
+import com.bagi.flowkotlin.model.Card
 import kotlinx.android.synthetic.main.activity_main.*
 
 class Covid19Activity : AppCompatActivity() {
     private lateinit var covid19ActivityViewModel: Covid19ActivityViewModel
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        val binding = DataBindingUtil.setContentView<ActivityMainBinding>(this, R.layout.activity_main)
-        binding.activity = this
+        val binding =
+            DataBindingUtil.setContentView<ActivityMainBinding>(this, R.layout.activity_main)
+        binding.viewModel = Covid19ActivityViewModel()
```

Comment:
If you think to bind view model instead view (the activity) why you're create new refernce to view model, you can use with covid19ActivityViewModel paramter.

Reply:
I think it is a bad practice to reference the activity in the xml. this is way i deleted the activty and instead constructed the viewmodel object. this is all you need.
Please go to these reference*: https://developer.android.com/topic/libraries/data-binding/architecture#viewmodel

* google have an example for that.

Reply:
ok I understand that, but you can use with covid19ActivityViewModel paramter that declare at the top of activity, and not create a new reference to Covid19ActivityViewModel

Reply:
You are absolutely right! Fixed it :)